doc/maintenance.md

```diff
@@ -231,7 +231,7 @@ publishing the actual release notes at {doc}`changes`.
 
 ### Check the README Syntax
 
-GitHub is a bit forgiving when it comes to the RST syntax in the README but PyPI is not.
+GitHub is a bit forgiving when it comes to the syntax in the README but PyPI is not.
 To check the README syntax, visit the
 [PyGMT TestPyPI release history](https://test.pypi.org/project/pygmt/#history), select
 the latest commit, and review the left sidebar and project description for any errors.
```

Comment:
Do we want to keep this subsection in the Maintainers Guide as a reminder (for new maintainers / contributors), or do people think we can remove it completely?

Reply:
I think we should still keep it, since there are also many Markdown flavors.

Reply:
I am fine with keeping this subsection in the Maintainers Guide!